piecrust/src/store/commit/reader.rs

```diff
@@ -0,0 +1,310 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::store::baseinfo::BaseInfo;
+use crate::store::commit::Commit;
+use crate::store::commit_store::CommitStore;
+use crate::store::hasher::Hash;
+use crate::store::index::{ContractIndexElement, NewContractIndex};
+use crate::store::tree::{position_from_contract, ContractsMerkle};
+use crate::store::treepos::TreePos;
+use crate::store::{
+    Bytecode, ContractSession, Module, BASE_FILE, BYTECODE_DIR, LEAF_DIR,
+    MAIN_DIR, MEMORY_DIR, OBJECTCODE_EXTENSION, TREE_POS_FILE,
+    TREE_POS_OPT_FILE,
+};
+use dusk_wasmtime::Engine;
+use piecrust_uplink::ContractId;
+use std::collections::BTreeMap;
+use std::fs::OpenOptions;
+use std::io::BufReader;
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+use std::{fs, io};
+
+pub struct CommitReader;
+
+impl CommitReader {
+    ///
+    /// Reads all commits into a given commit store
+    pub fn read_all_commits<P: AsRef<Path>>(
+        engine: &Engine,
+        root_dir: P,
+        commit_store: Arc<Mutex<CommitStore>>,
+    ) -> io::Result<()> {
+        let root_dir = root_dir.as_ref();
+
+        let root_dir = root_dir.join(MAIN_DIR);
+        fs::create_dir_all(&root_dir)?;
+
+        for entry in fs::read_dir(root_dir)? {
+            let entry = entry?;
+            if entry.path().is_dir() {
+                let filename = entry.file_name();
+                if filename == MEMORY_DIR
+                    || filename == BYTECODE_DIR
+                    || filename == LEAF_DIR
+                {
+                    continue;
+                }
+                tracing::trace!("before read_commit");
+                let commit = Self::commit_from_dir(
+                    engine,
+                    entry.path(),
+                    commit_store.clone(),
+                )?;
+                tracing::trace!("before read_commit");
```

Comment:
This should be "after", right?

```suggestion
                tracing::trace!("after read_commit");
```

Reply:
Done